csharp/SecureMemory/SecureMemory/ProtectedMemoryImpl/Linux/LinuxOpenSSL11ProtectedMemoryAllocatorLP64.cs

```diff
@@ -1,38 +1,37 @@
 using System;
 using System.Diagnostics;
 using System.Runtime.CompilerServices;
-using System.Runtime.InteropServices;
-using System.Text;
-using System.Xml.Schema;
-using GoDaddy.Asherah.PlatformNative.LP64.Libc;
 using GoDaddy.Asherah.PlatformNative.LP64.Linux;
-using GoDaddy.Asherah.SecureMemory.ProtectedMemoryImpl.Linux;
+using Microsoft.Extensions.Configuration;
 
 [assembly: InternalsVisibleTo("SecureMemory.Tests")]
 [assembly: InternalsVisibleTo("DynamicProxyGenAssembly2")]
 
 namespace GoDaddy.Asherah.SecureMemory.ProtectedMemoryImpl.Linux
 {
-    internal class LinuxOpenSSL11ProtectedMemoryAllocatorLP64 : LinuxProtectedMemoryAllocatorLP64, IProtectedMemoryAllocator, IDisposable
+    internal class LinuxOpenSSL11ProtectedMemoryAllocatorLP64 : LinuxProtectedMemoryAllocatorLP64
     {
         private readonly ulong blockSize;
         private LinuxOpenSSL11LP64 openSSL11;
         private OpenSSLCryptProtectMemory cryptProtectMemory;
         private bool disposedValue;
 
-        public LinuxOpenSSL11ProtectedMemoryAllocatorLP64(ulong size, int minsize)
-            : base((LinuxLibcLP64)new LinuxOpenSSL11LP64())
+        public LinuxOpenSSL11ProtectedMemoryAllocatorLP64(IConfiguration configuration)
+            : base(new LinuxOpenSSL11LP64())
         {
             openSSL11 = (LinuxOpenSSL11LP64)GetLibc();
             if (openSSL11 == null)
             {
                 throw new Exception("GetLibc returned null object for openSSL11");
             }
 
+            ulong heapSize = ulong.Parse(configuration["heapSize"]);
+            int minimumAllocationSize = int.Parse(configuration["minimumAllocationSize"]);
```

Comment:
We should validate the configuration to prevent NRE and missing/invalid settings. Maybe we could add sensible defaults here like you've done in the windows allocator?

For example, dropping the following into `ProtectedMemorySecretFactoryTest.cs` results in an ArgumentNullException on L28

```csharp
        [Fact]
        private void TestInvalidConfiguration()
        {
            var configuration = new ConfigurationBuilder().AddInMemoryCollection(new Dictionary<string, string>()
            {
                {"secureHeapEngine", "openssl11"}  // Note the missing "required" settings
            }).Build();

            Debug.WriteLine("ProtectedMemorySecretFactoryTest.TestInvalidConfiguration");
            using (var factory = new ProtectedMemorySecretFactory(configuration))
            {
            }
        }
```

Reply:
Agreed